agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java

```diff
@@ -450,6 +461,45 @@ private static String overlayConnectionStringFromEnv(String connectionString) {
     return null;
   }
 
+  private static Configuration.Proxy overlayProxyFromEnv(Configuration.Proxy proxy)
+      throws MalformedURLException, UnsupportedEncodingException {
+
+    String proxyEnvVar = getEnvVar(APPLICATIONINSIGHTS_PROXY);
+    if (proxyEnvVar == null) {
+      if (proxy.password != null) {
+        configurationLogger.warn(
+            "Storing the proxy password in the application insights json configuration file"
+                + " is deprecated because it is not secure. Please use the"
+                + " APPLICATIONINSIGHTS_PROXY environment variable instead which supports passing"
+                + " the username and password, e.g."
+                + " APPLICATIONINSIGHTS_PROXY=https://myuser:mypassword@myproxy:8888");
+      }
+      return proxy;
+    }
+
+    Configuration.Proxy proxyFromEnv = new Configuration.Proxy();
+
+    URL proxyUrl = new URL(proxyEnvVar);
+    proxyFromEnv.host = proxyUrl.getHost();
+    proxyFromEnv.port = proxyUrl.getPort();
+    if (proxyFromEnv.port == -1) {
+      proxyFromEnv.port = proxyUrl.getDefaultPort();
+    }
+
+    String userInfo = proxyUrl.getUserInfo();
+    if (userInfo != null) {
+      String[] usernamePassword = userInfo.split(":", 2);
+      if (usernamePassword.length == 2) {
+        proxyFromEnv.username =
+            URLDecoder.decode(usernamePassword[0], StandardCharsets.UTF_8.toString());
+        proxyFromEnv.password =
+            URLDecoder.decode(usernamePassword[1], StandardCharsets.UTF_8.toString());
+      }
```

Comment:
should we log warning when applicationinsights_proxy is malformatted?

Reply:
done